tokio/src/sync/semaphore.rs

```diff
@@ -968,6 +968,25 @@ impl<'a> SemaphorePermit<'a> {
     /// Forgets the permit **without** releasing it back to the semaphore.
     /// This can be used to reduce the amount of permits available from a
     /// semaphore.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # fn main() {
```

Comment:
You can remove this `fn main()` wrapper. The doc-test framework adds them automatically. We only need our own when we need to `.await` something.